pkg/apic/provisioning/requeststatus.go

```diff
@@ -0,0 +1,69 @@
+package provisioning
+
+import "fmt"
+
+const statusSetError = "can not set status as it's already been set"
+
+// RequestStatus - holds info about the status of the request
+type RequestStatus struct {
+	status  status
+	message string
+}
+
+// RequestStatusBuilder - builder to create new request status
+type RequestStatusBuilder interface {
```

Comment:
I have the same feedback on this builder too. But, I don't see enough of how it is meant to be used from the tests to know how it is meant to be used.

Reply:
```
const (
	// Success - request was successful
	Success status = iota + 1
	// Failed - request failed
	Failed
)

type RequestStatus struct {
	status  status
	message string
}

func NewSuccessStatus() RequestStatus {
	return RequestStatus{
		status:  Success,
		message: "",
	}
}

func NewErrorStatus(message string) RequestStatus {
	return RequestStatus{
		status:  Failed,
		message: message,
	}
}
```

Reply:
This avoids error checking and a processing func that is needed with the builder.

Reply:
The agents will use both builders to create their respective types.  These types will be used as return data for the provision or deprovision method requests

Reply:
After Vivek's feedback and subsequent updates this may be better shown.  The credential and requestStatus types are meant to be SDK only types.  So the agent implementation will not have direct access to their values.  The builders have been updated to return only interfaces and the concrete types are private now.